From b4599ff0dd03838453343b0df5170588c58b54e5 Mon Sep 17 00:00:00 2001
From: Johnathan Mantey <johnathanx.mantey@intel.com>
Date: Tue, 28 May 2024 12:14:16 -0700
Subject: [PATCH] Write IPv4 default gateway configuration

This is a backport of af38fe9dd741c0b8368edb69d05e233e8a9b5223
phosphor-networkd: write down the config when GW changes

The default gateway D-Bus entry was not being copied into the
systemd-networkd configuration file after it had been updated.

Signed-off-by: Johnathan Mantey <johnathanx.mantey@intel.com>
---
 src/ethernet_interface.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index dd938e2..97bdc69 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -882,6 +882,7 @@ std::string EthernetInterface::defaultGateway(std::string gateway)
     if (gateway != defaultGateway())
     {
         gateway = EthernetInterfaceIntf::defaultGateway(std::move(gateway));
+        writeConfigurationFile();
         manager.get().reloadConfigs();
     }
     return gateway;
@@ -893,6 +894,7 @@ std::string EthernetInterface::defaultGateway6(std::string gateway)
     if (gateway != defaultGateway6())
     {
         gateway = EthernetInterfaceIntf::defaultGateway6(std::move(gateway));
+        writeConfigurationFile();
         manager.get().reloadConfigs();
     }
     return gateway;
-- 
2.45.1

